spyql/processor.py

```diff
@@ -409,18 +468,25 @@ def get_input_iterator(self):
 
 
 class TextProcessor(Processor):
-    def __init__(self, prs, strings):
-        super().__init__(prs, strings)
+    def __init__(self, prs, strings, interactive, filepath = None):
+        super().__init__(prs, strings, interactive)
+        self.filepath = filepath
 
     # reads a text row as a row with 1 column
     def get_input_iterator(self):
         # to do: suport files
-        return ([line.rstrip("\n\r")] for line in sys.stdin)
+        if self.interactive:
+            with open(self.filepath, "r") as f:
+                for line in f:
+                    # note yeilding to avoid reading in one go
+                    yield [line.rstrip("\n\r")]
```

Comment:
I believe this is equivalent to using a generator like this (but never compared performance):
`([line.rstrip("\n\r")] for line in f)`

Reply:
Cannot do this because it throws the error:
```
  File "/Users/yashbonde/Desktop/ai/repos/spyql/spyql/processor.py", line 525, in <genexpr>
    return ([decoder.decode(line)] for line in f)
ValueError: I/O operation on closed file.
```

Reply:
This would have to happen inside the `with` block... but again, seems more a matter of style and having more compact code. I will revisit this when I review your latest changes

Reply:
I did implement it in a `with` like follows but threw the error as above, maybe we can leave it as is for now:

```
with open(...):
  return (...)
```